webapp/src/components/root/root.jsx

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import {makeStyleFromTheme, changeOpacity} from 'mattermost-redux/utils/theme_utils';
+import {Link} from 'react-router-dom'
+import {isDirectChannel} from 'mattermost-redux/utils/channel_utils';
+import {Client4} from 'mattermost-redux/client';
+import {getConfig} from 'mattermost-redux/selectors/entities/general';
+import {id as pluginId} from '../../manifest';
+
+const {Tooltip, Popover, OverlayTrigger, Modal} = window.ReactBootstrap
+
+export default class Root extends React.PureComponent {
+  static propTypes = {
```

Comment:
nit: Not needed to follow, but at mattermost we normally use 4 spaces for javascript. Unless any other core committer have an strong opinion here, feel free to choose follow our styling or not.

Reply:
@hanzei has an `.editorconfig` ready for the sample plugin that might help here -- recommend merging https://github.com/mattermost/mattermost-plugin-sample/pull/34 as such.

Reply:
I added the `.editorconfig` and will try to update all the files to follow the MM standards.